src/box/lua/net_box.c

```diff
@@ -2053,22 +2041,16 @@ netbox_iproto_auth(struct lua_State *L)
 		return luaL_error(L, "Invalid salt");
 	struct netbox_transport *transport = luaT_check_netbox_transport(L, 4);
 	int fd = lua_tointeger(L, 5);
-	if (netbox_encode_auth(&transport->send_buf, transport->next_sync++,
-			       user, user_len, password, password_len,
-			       salt) != 0) {
-		goto error;
-	}
+	netbox_encode_auth(L, &transport->send_buf, transport->next_sync++,
+			   user, user_len, password, password_len, salt);
 	struct xrow_header hdr;
 	if (netbox_send_and_recv(transport, fd, &hdr) != 0)
-		goto error;
+		return luaT_error(L);
```

Comment:
It is consistent: functions that deal with a Lua state raise errors while low-level functions that don't use a Lua state return -1.

The reason why I'm doing this is that this change makes it easier to convert everything to C. We have two options here:
1. Either wrap all Lua code pieces in pcalls and return -1 if any of them fails.
2. Wrap the whole state machine code in a pcall.

The latter seems easier and requires less updates of the existing functions. Take `netbox_transport_dispatch_response` for example. It calls some Lua functions and creates some Lua objects - wrapping all those pieces in pcalls would make the code look convoluted IMO, it's easier to wrap the whole function in a pcall. Then, this function is called by `netbox_iproto_schema`, which also pushes some values to a Lua stack, so going further, we can wrap the whole `netbox_iproto_schema` function in a pcall to keep the code linear and simple, and so on.

I'm currently working on the patch converting the state machine to C. I'll update this patch set soon with it so that you can see the whole picture.